daemon/src/bitmex_price_feed.rs

```diff
@@ -36,184 +65,172 @@ impl Actor {
+    pub bin_size: BinSize,
+    /// Instrument symbol.
+    pub symbol: String,
+    /// Number of results to fetch.
+    pub count: u32,
+    /// If true, will sort results newest first.
+    pub reverse: bool,
 }
 
-impl Quote {
-    fn from_str(text: &str) -> Result<Option<Self>> {
-        let table_message = match serde_json::from_str::<wire::TableMessage>(text) {
-            Ok(table_message) => table_message,
-            Err(_) => {
-                tracing::trace!(%text, "Not a 'table' message, skipping...");
-                return Ok(None);
-            }
-        };
-
-        let [quote] = table_message.data;
-
-        Ok(Some(Self {
-            timestamp: Timestamp::parse_from_rfc3339(&quote.timestamp)?,
-            bid: Price::new(Decimal::try_from(quote.bid_price)?)?,
-            ask: Price::new(Decimal::try_from(quote.ask_price)?)?,
-        }))
-    }
-
-    pub fn for_maker(&self) -> Price {
-        self.ask
+impl GetQuoteBucketedRequest {
+    fn latest_xbt_usd() -> Self {
+        Self {
+            bin_size: BinSize::M1,
+            symbol: "XBTUSD".to_string(),
+            count: 1,
+            reverse: true,
+        }
     }
+}
 
-    pub fn for_taker(&self) -> Price {
-        self.mid_range()
-    }
+#[derive(Debug, Clone, Copy, Serialize)]
+#[serde(rename = "XBTUSD")]
+struct XbtUsd;
 
-    fn mid_range(&self) -> Price {
-        (self.bid + self.ask) / 2
-    }
+#[derive(Clone, Debug, Serialize)]
+enum BinSize {
+    #[serde(rename = "1m")]
+    M1,
+}
 
-    pub fn is_older_than(&self, duration: time::Duration) -> bool {
-        let required_quote_timestamp = (OffsetDateTime::now_utc() - duration).unix_timestamp();
+// The error response from bitmex;
+#[derive(Deserialize, Debug, Clone)]
+struct BitmexErrorResponse {
+    error: BitmexErrorMessage,
+}
 
-        self.timestamp.seconds() < required_quote_timestamp
-    }
+#[derive(Deserialize, Debug, Clone)]
+struct BitmexErrorMessage {
+    message: String,
+    name: String,
 }
 
-mod wire {
-    use serde::Deserialize;
+#[derive(Default)]
+struct Client(reqwest::Client);
 
-    #[derive(Debug, Clone, Deserialize, PartialEq)]
-    pub struct TableMessage {
-        pub table: String,
-        // we always just expect a single quote, hence the use of an array instead of a vec
-        pub data: [QuoteData; 1],
+impl Client {
+    fn new() -> Self {
+        Self(reqwest::Client::new())
     }
 
-    #[derive(Debug, Clone, Deserialize, PartialEq)]
-    #[serde(rename_all = "camelCase")]
-    pub struct QuoteData {
-        pub bid_size: u64,
-        pub ask_size: u64,
-        pub bid_price: f64,
-        pub ask_price: f64,
-        pub symbol: String,
-        pub timestamp: String,
+    async fn get_latest_quote(&self) -> Result<Quote> {
+        let mut url = reqwest::Url::parse(BITMEX_QUOTE_URL).expect("valid URL");
+        let query = serde_urlencoded::to_string(GetQuoteBucketedRequest::latest_xbt_usd())
+            .expect("valid query");
```

Comment:
I'd probably just hardcode this URL with is query parameters, same as we did before!

Reply:
You should then be able to just use `reqwest::get`.